third_party/terraform/utils/config.go.erb

```diff
@@ -668,7 +668,11 @@ func (c *Config) getTokenSource(clientScopes []string) (oauth2.TokenSource, erro
 
 	log.Printf("[INFO] Authenticating using DefaultClient...")
 	log.Printf("[INFO]   -- Scopes: %s", clientScopes)
-	return googleoauth.DefaultTokenSource(context.Background(), clientScopes...)
+	if c, err := googleoauth.DefaultTokenSource(context.Background(), clientScopes...); err == nil {
+		return c, err
+	} else {
+		return c, fmt.Errorf("Attempted to load application default credentials since neither `credentials` nor `access_token` was set in the provider block.  No credentials loaded.  Consider running 'gcloud auth application-default login.  Original error: %w", err)
```

Comment:
```suggestion
		return c, fmt.Errorf("Attempted to load application default credentials since neither `credentials` nor `access_token` was set in the provider block.  No credentials loaded.  Consider running 'gcloud auth application-default login'.  Original error: %w", err)
```